packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -272,343 +314,216 @@ function hasNonLinkText(device: Device): Predicate<Element> {
- * a better solution.
- */
-
-function hasDistinguishableFont(
-  container: Element,
-  device: Device,
-  context?: Context
-): Predicate<Element> {
-  const style = Style.from(container, device, context);
-
-  const referenceWeight = style.computed("font-weight").value;
-  const referenceFamily = Option.from(
-    style.computed("font-family").value.values[0]
-  );
-
-  return or(
-    hasComputedStyle(
-      "font-weight",
-      not((weight) => weight.equals(referenceWeight)),
-      device,
-      context
-    ),
-    hasComputedStyle(
-      "font-family",
-      not((family) => Option.from(family.values[0]).equals(referenceFamily)),
-      device,
-      context
-    )
-  );
-}
-
-function hasDistinguishableVerticalAlign(
-  container: Element,
-  device: Device,
-  context?: Context
-): Predicate<Element> {
-  const reference = Style.from(container, device, context).computed(
-    "vertical-align"
-  ).value;
-
-  return hasComputedStyle(
-    "vertical-align",
-    not((alignment) => alignment.equals(reference)),
-    device,
-    context
-  );
-}
-
-function hasDistinguishableCursor(
-  container: Element,
-  device: Device,
-  context?: Context
-): Predicate<Element> {
-  // Checking if there is a custom cursor, otherwise grabbing the built-in
-  function getFirstCursor(style: Style.Computed<"cursor">) {
-    return style.values[0].values.length !== 0
-      ? style.values[0].values[0]
-      : style.values[1];
-  }
-  const containerCursorStyle = Style.from(container, device, context).computed(
-    "cursor"
-  ).value;
-
-  // We assume that the first custom cursor, if any, will never fail to load
-  // and thus don't try to default further.
-  const reference = getFirstCursor(containerCursorStyle);
-
-  return hasComputedStyle(
-    "cursor",
-    not((cursor) => getFirstCursor(cursor).equals(reference)),
-    device,
-    context
-  );
-}
-
-type Name = Property.Name | Property.Shorthand.Name;
-
-export class ComputedStyles implements Equatable, Hashable, Serializable {
-  public static of(
-    style: Iterable<readonly [Name, string]> = []
-  ): ComputedStyles {
-    return new ComputedStyles(Map.from(style));
-  }
-
-  private readonly _style: Map<Name, string>;
-
-  private constructor(style: Map<Name, string>) {
-    this._style = style;
-  }
-
-  public get style(): Map<Name, string> {
-    return this._style;
+namespace Distinguishable {
+  export function isDistinguishable(
+    container: Element,
+    device: Device,
+    context: Context = Context.empty()
+  ): Predicate<Element> {
+    return or(
+      // Things like text decoration and backgrounds risk blending with the
+      // container element. We therefore need to check if these can be distinguished
+      // from what the container element might itself set.
+      hasDistinguishableContrast(container, device, context),
+      hasDistinguishableBackground(container, device, context),
```

Comment:
This is a bit fragile, because of the caching and calculating the contrast pairings once, `hasDistinguishableContrast` has to be the first one otherwise some tests fail. My theory is that if anything else is "true" that's calculated before the things after won't be checked.